WordPress/Classes/ViewRelated/Reader/ReaderBrowseSiteViewController.m

```diff
@@ -0,0 +1,206 @@
+#import "ReaderBrowseSiteViewController.h"
+
+#import <WordPress-iOS-Shared/WPStyleGuide.h>
+
+#import "AccountService.h"
+#import "ContextManager.h"
+#import "CustomHighlightButton.h"
+#import "ReaderPost.h"
+#import "ReaderPostService.h"
+#import "ReaderPostsViewController.h"
+#import "ReaderPostDetailViewController.h"
+#import "ReaderPostAttributionView.h"
+#import "ReaderSubscriptionViewController.h"
+#import "ReaderPreviewHeaderView.h"
+#import "ReaderSiteHeaderView.h"
+#import "ReaderTopic.h"
+#import "ReaderTopicService.h"
+#import "UIImageView+AFNetworkingExtra.h"
+#import "WPTabBarController.h"
+#import "WPWebViewController.h"
+#import "WordPress-Swift.h"
+
+@interface ReaderBrowseSiteViewController ()<WPContentAttributionViewDelegate>
+@property (nonatomic, strong) ReaderPostsViewController *postsViewController;
+@property (nonatomic, strong) ReaderSiteHeaderView *siteHeaderView;
+@property (nonatomic, strong) ReaderPreviewHeaderView *tableHeaderView;
+@property (nonatomic, strong) ReaderPost *post;
+@property (nonatomic, strong) ReaderTopic *siteTopic;
+@end
+
+@implementation ReaderBrowseSiteViewController
+
+#pragma mark - Lifecycle Methods
+
+- (void)dealloc
+{
+    NSManagedObjectContext *context = [[ContextManager sharedInstance] mainContext];
+    ReaderTopicService *service = [[ReaderTopicService alloc] initWithManagedObjectContext:context];
+    [service deleteTopic:self.siteTopic];
```

Comment:
Nuking CoreData entities in the dealloc (feels) a little bit risky. Would it be possible to move this to `viewWillDisappear` perhaps?